tokio/src/runtime/task/join.rs

```diff
@@ -17,7 +17,8 @@ doc_rt_core! {
     /// on it.
     ///
     /// This `struct` is created by the [`task::spawn`] and [`task::spawn_blocking`]
-    /// functions.
+    /// functions. The struct is parameterized over the return type of the spawned
+    /// task (which is a `Result<T, JoinError>`).
```

Comment:
How about something like this?

> This `struct` is created by the [`task::spawn`] and [`task::spawn_blocking`]functions. The struct is parameterized over the return type of the spawned task. The return type of the `handle.await` is a `Result<T, JoinError>`.

If we don't mention the return type of the `handle.await`, then, I feel the later example with the Result being returned from a task, might raise more questions.

Reply:
We should definitely mention the return type of `handle.await`, I just think that they are "too close" to each other in some sense, and that it is too easy to confuse them.

As for your suggested rewording, I think it could be made even clearer. E.g. maybe:

> The generic parameter `T` in `JoinHandle<T>` is the return type of the spawned task.
>
> ... code example with `JoinHandle<i32>` that assigns `handle.await` to a variable of type result 
>
> Note that `handle.await` doesn't give you the return type directly — it is wrapped in a `Result` because panics in the spawned task are caught by Tokio. If the task panics, the error is a `JoinError` that contains the panic.